cupy/fft/fft.py

```diff
@@ -395,22 +402,23 @@ def _fftn(a, s, axes, norm, direction, value_type='C2C', order='A', plan=None,
     # sort the provided axes in ascending order
     axes = tuple(sorted(np.mod(axes, a.ndim)))
 
+    xp = cupy.get_array_module(a)
```

Comment:
Why do you want to support for numpy.ndarray inputs? Other CuPy's routines do not accept numpy.ndarray as input basically.

Reply:
This is because CuPy's internals take too much device memory. For example, as mentioned in the revised PR description, it's impossible to FFT a 8G array on one P100. One would think 2 P100s would help (which has 32G memory in total), but it turns out not possible either. 

But, starting from the host directly, I can FFT a 12G array without problem on 2 P100s. I can even try a 32G array on 2 V100s if you wish. I think this is the most important advantage of multi-GPU FFT.

> Other CuPy's routines do not accept numpy.ndarray as input basically.

True, but other CuPy routines do not have NVIDIA's native multi-GPU support either 🙂 so whenever there is support, I think we should take advantage of it.

As I said, we could either keep this "NumPy-in, NumPy-out" feature undocumented until I add support of multi-GPU `PlanNd`, or mark it experimental. Either is better than nothing.

Reply:
Could you split this fix to another PR? It is certainly related to this PR, but it is over the scope of multi-GPU support.

Reply:
Sorry, what fix were you referring to? And if you don't want `set_cufft_gpus()`, anything changed in `cupy/fft/fft.py` is meaningless and should be undone, as high-level APIs need not be touched at all.

Reply:
Sorry for the confusion. The above comment refers to `numpy.ndarray.` support.
How about supporting multi-GPU `Plan1d` in this PR, and `numpy.ndarray` input in another PR? We need to discuss whether to support numpy inputs, and it will take a long time. So I want to split the discussion into 2 PRs and finish multi-GPU support first.

Reply:
For the `numpy.ndarray` support, may I suggest instead of completely wiping it out from the codebase, we simply comment it out and leave a TODO?

Reply:
The discussion for NumPy support is in #2742.

Reply:
@asi1024, how about my suggestion https://github.com/cupy/cupy/pull/2644#discussion_r352445583 above?

Reply:
Do you mean that you will remove the "NumPy-in, NumPy-out" from the current code? If so, I agree with your suggestion.